packages/flutter_tools/lib/src/commands/update_packages.dart

```diff
@@ -60,3 +102,418 @@ class UpdatePackagesCommand extends FlutterCommand {
     printStatus('\nRan \'pub\' $count time${count == 1 ? "" : "s"} and fetched coverage data in ${seconds.toStringAsFixed(1)}s.');
   }
 }
+
+enum Section { dependencies, devDependencies, dependencyOverrides, other }
+
+const String kTransitiveMagicString = '# TRANSITIVE DEPENDENCY';
+
+class PubspecYaml {
+  factory PubspecYaml(Directory directory) {
+    final File file = _pubspecFor(directory);
+    return new PubspecYaml._(file, _parse(file.path, file.readAsLinesSync()));
+  }
+
+  PubspecYaml._(this.file, this.inputData);
+
+  final File file;
+
+  final List<PubspecLine> inputData;
+
+  static List<PubspecLine> _parse(String filename, List<String> lines) {
+    final List<PubspecLine> result = <PubspecLine>[];
+    Section section = Section.other;
```

Comment:
This value changes are you scan the file, and indicates the section the scanner is currently in. A better name might be `currentSection`.

Reply:
added comments